diff --git a/src/common/lib/ckeditor/lang/da.js b/src/common/lib/ckeditor/lang/da.js
index a2d613e..b5cae6f 100644
--- a/src/common/lib/ckeditor/lang/da.js
+++ b/src/common/lib/ckeditor/lang/da.js
@@ -155,10 +155,10 @@
         "cut": "Klip",
         "cutError": "Din browsers sikkerhedsindstillinger tillader ikke editoren at få automatisk adgang til udklipsholderen.<br><br>Brug i stedet tastaturet til at klippe teksten (Ctrl/Cmd+X).",
         "paste": "Indsæt",
-        "pasteNotification": "Press %1 to paste. Your browser doesn‘t support pasting with the toolbar button or context menu option.",
-        "pasteArea": "Paste Area",
-        "pasteMsg": "Paste your content inside the area below and press OK.",
-        "title": "Paste"
+        "pasteNotification": "Tast %1 for at indsætte. Din browser understøtter ikke indsætning via knappen i værktøjslinjen eller som valg fra menu.",
+        "pasteArea": "Insætningsområde",
+        "pasteMsg": "Indsæt indhold indenfor området nedenfor og klik OK.",
+        "title": "Indsæt"
     },
     "indent": {
         "indent": "Forøg indrykning",
@@ -258,13 +258,13 @@
     },
     "font": {
         "fontSize": {
-            "label": "Size",
-            "voiceLabel": "Font Size",
-            "panelTitle": "Font Size"
+            "label": "Størrelse",
+            "voiceLabel": "Skriftstørrelse",
+            "panelTitle": "Skriftstørrelse"
         },
-        "label": "Font",
-        "panelTitle": "Font Name",
-        "voiceLabel": "Font"
+        "label": "Skrifttype",
+        "panelTitle": "Skrifttypenavn",
+        "voiceLabel": "Skrift"
     },
     "format": {
         "label": "Formatering",
@@ -281,12 +281,12 @@
         "tag_pre": "Formateret"
     },
     "format_buttons": {
-        "h1": "Heading 1",
-        "h2": "Heading 2",
-        "h3": "Heading 3",
-        "h4": "Heading 4",
-        "h5": "Heading 5",
-        "h6": "Heading 6"
+        "h1": "Overskrift 1",
+        "h2": "Overskrift 2",
+        "h3": "Overskrift 3",
+        "h4": "Overskrift 4",
+        "h5": "Overskrift 5",
+        "h6": "Overskrift 6"
     },
     "iframe": {
         "border": "Vis kant på rammen",
@@ -310,8 +310,8 @@
     },
     "pastetext": {
         "button": "Indsæt som ikke-formateret tekst",
-        "pasteNotification": "Your browser does not allow you to paste plain text this way. Press %1 to paste.",
-        "title": "Paste as Plain Text"
+        "pasteNotification": "Din browser understøtter ikke indsætning at indsætte almindelig tekst på denne måde. Tast %1 for at indsætte.",
+        "title": "Indsæt som almindelig tekst"
     },
     "table": {
         "border": "Rammebredde",
@@ -384,7 +384,7 @@
         "widthUnit": "Bredde på enhed"
     },
     "quicktable": {
-        "more": "More..."
+        "more": "Mere..."
     },
     "removeformat": {
         "toolbar": "Fjern formatering"
@@ -420,12 +420,12 @@
             "000": "Sort",
             "800000": "Mørkerød",
             "8B4513": "Mørk orange",
-            "2F4F4F": "Dark Slate Grå",
-            "008080": "Teal",
-            "000080": "Navy",
+            "2F4F4F": "Skifergrå",
+            "008080": "Grønblå",
+            "000080": "Mørkeblå",
             "4B0082": "Indigo",
             "696969": "Mørkegrå",
-            "B22222": "Scarlet / Rød",
+            "B22222": "Skarlagenrød",
             "A52A2A": "Brun",
             "DAA520": "Guld",
             "006400": "Mørkegrøn",
@@ -437,7 +437,7 @@
             "FF8C00": "Mørk orange",
             "FFD700": "Guld",
             "008000": "Grøn",
-            "0FF": "Cyan",
+            "0FF": "Akvamarin",
             "00F": "Blå",
             "EE82EE": "Violet",
             "A9A9A9": "Matgrå",
@@ -447,45 +447,45 @@
             "00FF00": "Lime",
             "AFEEEE": "Mat tyrkis",
             "ADD8E6": "Lyseblå",
-            "DDA0DD": "Plum",
+            "DDA0DD": "Blomme",
             "D3D3D3": "Lysegrå",
-            "FFF0F5": "Lavender Blush",
+            "FFF0F5": "Lys lavendel",
             "FAEBD7": "Antikhvid",
             "FFFFE0": "Lysegul",
-            "F0FFF0": "Gul / Beige",
+            "F0FFF0": "Beige",
             "F0FFFF": "Himmeblå",
             "F0F8FF": "Alice Blue",
             "E6E6FA": "Lavendel",
             "FFF": "Hvid",
-            "1ABC9C": "Strong Cyan",
-            "2ECC71": "Emerald",
-            "3498DB": "Bright Blue",
-            "9B59B6": "Amethyst",
-            "4E5F70": "Grayish Blue",
-            "F1C40F": "Vivid Yellow",
-            "16A085": "Dark Cyan",
-            "27AE60": "Dark Emerald",
-            "2980B9": "Strong Blue",
-            "8E44AD": "Dark Violet",
-            "2C3E50": "Desaturated Blue",
+            "1ABC9C": "Kraftig akvamarin",
+            "2ECC71": "Smaragdgrøn",
+            "3498DB": "Klar Blå",
+            "9B59B6": "Ametyst",
+            "4E5F70": "Gråblå",
+            "F1C40F": "Lysende gul",
+            "16A085": "Mørk akvamarin",
+            "27AE60": "Mørk smaragdgrøn",
+            "2980B9": "Kraftig blå",
+            "8E44AD": "Mørkelilla",
+            "2C3E50": "Svagt blå",
             "F39C12": "Orange",
-            "E67E22": "Carrot",
-            "E74C3C": "Pale Red",
-            "ECF0F1": "Bright Silver",
-            "95A5A6": "Light Grayish Cyan",
-            "DDD": "Light Gray",
-            "D35400": "Pumpkin",
-            "C0392B": "Strong Red",
-            "BDC3C7": "Silver",
-            "7F8C8D": "Grayish Cyan",
-            "999": "Dark Gray"
+            "E67E22": "Gulerod",
+            "E74C3C": "Blegrød",
+            "ECF0F1": "Klar sølv",
+            "95A5A6": "Lys grålig akvamarin",
+            "DDD": "Lysegrå",
+            "D35400": "Græskar",
+            "C0392B": "Kraftig rød",
+            "BDC3C7": "Sølv",
+            "7F8C8D": "Grålig akvamarin",
+            "999": "Mørkegrå"
         },
         "more": "Flere farver...",
         "panelTitle": "Farver",
         "textColorTitle": "Tekstfarve"
     },
     "infoWidget": "Info widget",
-    "infoWidgetSelect": "Choose type of info-box",
-    "infoWidgetEdit": "Edit type of info-box",
-    "edit": "Edit"
+    "infoWidgetSelect": "Vælg type af info-boks",
+    "infoWidgetEdit": "Redigér type af info-boks",
+    "edit": "Redigér"
 };
\ No newline at end of file
diff --git a/src/common/lib/ckeditor/plugins/image/lang/da.js b/src/common/lib/ckeditor/plugins/image/lang/da.js
index ad204c4..d369f5d 100644
--- a/src/common/lib/ckeditor/plugins/image/lang/da.js
+++ b/src/common/lib/ckeditor/plugins/image/lang/da.js
@@ -3,24 +3,24 @@
 For licensing, see LICENSE.md or https://ckeditor.com/legal/ckeditor-oss-license
 */
 CKEDITOR.plugins.setLang('image', 'da', {
-    alt: 'Alternative Text',
-    border: 'Border',
+    alt: 'Alternativ Tekst',
+    border: 'Kant',
     browse: 'Browse',
-    btnUpload: 'Send it to the Server',
-    button2Img: 'Do you want to transform the selected image button on a simple image?',
+    btnUpload: 'Send til serveren',
+    button2Img: 'Vil du omdanne den valgte billedknap på et simpelt billede?',
     hSpace: 'HSpace',
-    img2Button: 'Do you want to transform the selected image on a image button?',
-    infoTab: 'Image Info',
+    img2Button: 'Vil du omdanne det valgte billede på en billedknap?',
+    infoTab: 'Billedinfo',
     linkTab: 'Link',
-    lockRatio: 'Lock Ratio',
-    menu: 'Image Properties',
-    resetSize: 'Reset Size',
-    title: 'Image Properties',
-    titleButton: 'Image Button Properties',
+    lockRatio: 'Lås proportion',
+    menu: 'Billedegenskaber',
+    resetSize: 'Gendan størrelse',
+    title: 'Billedegenskaber',
+    titleButton: 'Billedknap egenskaber',
     upload: 'Upload',
-    urlMissing: 'Image source URL is missing.',
+    urlMissing: 'Billedkilde URL mangler.',
     vSpace: 'VSpace',
-    validateBorder: 'Border must be a whole number.',
-    validateHSpace: 'HSpace must be a whole number.',
-    validateVSpace: 'VSpace must be a whole number.'
+    validateBorder: 'Kant skal vær et helt tal.',
+    validateHSpace: 'HSpace skal være et helt tal.',
+    validateVSpace: 'VSpace skal være et helt tal.'
 });
\ No newline at end of file
diff --git a/src/common/lib/ckeditor/plugins/layoutmanager/lang/da.js b/src/common/lib/ckeditor/plugins/layoutmanager/lang/da.js
index 61b2613..3fadb3d 100644
--- a/src/common/lib/ckeditor/plugins/layoutmanager/lang/da.js
+++ b/src/common/lib/ckeditor/plugins/layoutmanager/lang/da.js
@@ -1,7 +1,7 @@
 CKEDITOR.plugins.setLang('layoutmanager', 'da', {
-    title: 'Layouts',
-    addLayoutDialogTitle: 'Choose layout',
-    manageLayoutDialogTitle: 'Replace layout with',
-    removeLayoutMenuLabel: 'Remove layout',
-    manageLayoutMenuLabel: 'Replace layout'
+    title: 'Layout',
+    addLayoutDialogTitle: 'Vælg layout',
+    manageLayoutDialogTitle: 'Erstat layout med',
+    removeLayoutMenuLabel: 'Fjern layout',
+    manageLayoutMenuLabel: 'Erstat layout'
 });
diff --git a/src/common/lib/ckeditor/plugins/lineheight/lang/da.js b/src/common/lib/ckeditor/plugins/lineheight/lang/da.js
index 318b39d..c98e377 100644
--- a/src/common/lib/ckeditor/plugins/lineheight/lang/da.js
+++ b/src/common/lib/ckeditor/plugins/lineheight/lang/da.js
@@ -1,3 +1,3 @@
 CKEDITOR.plugins.setLang('lineheight','da', {
-    title: 'Line Height'
+    title: 'Linjehøjde'
 } );
