Index: src/common/lib/ckeditor/plugins/liststyle/dialogs/liststyle.js
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
--- src/common/lib/ckeditor/plugins/liststyle/dialogs/liststyle.js	(date 1498202937000)
+++ src/common/lib/ckeditor/plugins/liststyle/dialogs/liststyle.js	(revision )
@@ -136,6 +136,9 @@
 								var nextItem = firstItem,
 									conseq = oldStart,
 									startNumber = isNaN( val ) ? 1 : val;
+
+                element.setStyle('counter-reset', 'item ' + startNumber);
+
 								while ( ( nextItem = nextItem.getNext( listItem ) ) && conseq++ ) {
 									if ( nextItem.getAttribute( 'value' ) == conseq )
 										nextItem.setAttribute( 'value', startNumber + conseq - oldStart );
@@ -155,10 +158,16 @@
 							},
 							commit: function( element ) {
 								var value = this.getValue();
-								if ( value )
+                var listStyleType = element.getStyle('list-style-type');
+                if (listStyleType) {
+                  element.removeClass(listStyleType);
+                }
+								if ( value ) {
+                  element.addClass(value);
-									element.setStyle( 'list-style-type', value );
+                  element.setStyle('list-style-type', value);
-								else
+                } else {
-									element.removeStyle( 'list-style-type' );
+                  element.removeStyle('list-style-type');
+                }
 							}
 						} ]
 					} ]
