diff --git a/src/common/lib/ckeditor/plugins/image/dialogs/image.js b/src/common/lib/ckeditor/plugins/image/dialogs/image.js
index 4753d2d..6a86033 100644
--- a/src/common/lib/ckeditor/plugins/image/dialogs/image.js
+++ b/src/common/lib/ckeditor/plugins/image/dialogs/image.js
@@ -194,7 +194,7 @@
 					var dialog = this.getDialog(),
 						value = '',
 						dimension = this.id == 'txtWidth' ? 'width' : 'height',
-						size = element.getAttribute( dimension );
+						size = element.getAttribute( dimension ) || element.$[dimension] ? element.$[dimension].toString() : null;
 
 					if ( size )
 						value = checkDimension( size, value );
@@ -488,7 +488,7 @@
 						padding: 0,
 						children: [ {
 							type: 'hbox',
-							widths: [ '280px', '110px' ],
+							widths: [ '340px', '0px', '50px' ],
 							align: 'right',
 							className: 'cke_dialog_image_url',
 							children: [ {
@@ -537,7 +537,11 @@
 								setup: function( type, element ) {
 									if ( type == IMAGE ) {
 										var url = element.data( 'cke-saved-src' ) || element.getAttribute( 'src' );
-										var field = this;
+                    // Disable field if image is base64 because otherwise it can crash the browser
+                    if (url.indexOf('data:') === 0) {
+                      this.disable();
+                    }
+                    var field = this;
 
 										this.getDialog().dontResetSize = true;
 
@@ -557,6 +561,23 @@
 								},
 								validate: CKEDITOR.dialog.validate.notEmpty( editor.lang.image.urlMissing )
 							},
+              {
+                type: 'file',
+                id: 'browseFile',
+                hidden: true,
+                setup: function() {
+                  this.getInputElement().$.setAttribute('accept', 'image/*');
+                },
+                onChange: function() {
+                  var reader = new FileReader();
+                  reader.addEventListener('load', () => {
+                    var urlElement = this.getDialog().getContentElement( 'info', 'txtUrl' );
+                    urlElement.disable();
+                    urlElement.setValue(reader.result);
+                  }, false);
+                  reader.readAsDataURL(this.getInputElement().$.files[0]);
+                }
+              },
 							{
 								type: 'button',
 								id: 'browse',
@@ -564,9 +585,11 @@
 								// TODO: We need something better than a fixed size here.
 								style: 'display:inline-block;margin-top:14px;',
 								align: 'center',
-								label: editor.lang.common.browseServer,
-								hidden: true,
-								filebrowser: 'info:txtUrl'
+								label: editor.lang.image.browse,
+                onClick: function() {
+                  var fileBrowseElement = this.getDialog().getContentElement( 'info', 'browseFile' );
+                  fileBrowseElement.getInputElement().$.click();
+                }
 							} ]
 						} ]
 					},
diff --git a/src/common/lib/ckeditor/plugins/image/lang/bg.js b/src/common/lib/ckeditor/plugins/image/lang/bg.js
index 5171738..9b33d74 100644
--- a/src/common/lib/ckeditor/plugins/image/lang/bg.js
+++ b/src/common/lib/ckeditor/plugins/image/lang/bg.js
@@ -5,6 +5,7 @@ For licensing, see LICENSE.md or http://ckeditor.com/license
 CKEDITOR.plugins.setLang( 'image', 'bg', {
 	alt: 'Алтернативен текст',
 	border: 'Рамка',
+  browse: 'Избери',
 	btnUpload: 'Изпрати я на сървъра',
 	button2Img: 'Do you want to transform the selected image button on a simple image?', // MISSING
 	hSpace: 'Хоризонтален отстъп',
diff --git a/src/common/lib/ckeditor/plugins/image/lang/en.js b/src/common/lib/ckeditor/plugins/image/lang/en.js
index 706bb9c..6669819 100644
--- a/src/common/lib/ckeditor/plugins/image/lang/en.js
+++ b/src/common/lib/ckeditor/plugins/image/lang/en.js
@@ -5,6 +5,7 @@ For licensing, see LICENSE.md or http://ckeditor.com/license
 CKEDITOR.plugins.setLang( 'image', 'en', {
 	alt: 'Alternative Text',
 	border: 'Border',
+  browse: 'Browse',
 	btnUpload: 'Send it to the Server',
 	button2Img: 'Do you want to transform the selected image button on a simple image?',
 	hSpace: 'HSpace',
