diff --git a/src/common/lib/ckeditor/plugins/layoutmanager/lang/de.js b/src/common/lib/ckeditor/plugins/layoutmanager/lang/de.js
new file mode 100644
index 0000000..3d4cbd3
--- /dev/null
+++ b/src/common/lib/ckeditor/plugins/layoutmanager/lang/de.js
@@ -0,0 +1,7 @@
+CKEDITOR.plugins.setLang('layoutmanager', 'de', {
+    title: 'Layouts',
+    addLayoutDialogTitle: 'Choose layout',
+    manageLayoutDialogTitle: 'Replace layout with',
+    removeLayoutMenuLabel: 'Remove layout',
+    manageLayoutMenuLabel: 'Replace layout'
+});
diff --git a/src/common/lib/ckeditor/plugins/layoutmanager/lang/fi.js b/src/common/lib/ckeditor/plugins/layoutmanager/lang/fi.js
new file mode 100644
index 0000000..591f484
--- /dev/null
+++ b/src/common/lib/ckeditor/plugins/layoutmanager/lang/fi.js
@@ -0,0 +1,7 @@
+CKEDITOR.plugins.setLang('layoutmanager', 'fi', {
+    title: 'Asettelut',
+    addLayoutDialogTitle: 'Valitse asettelu',
+    manageLayoutDialogTitle: 'Korvaa asettelu',
+    removeLayoutMenuLabel: 'Poista asettelu',
+    manageLayoutMenuLabel: 'Korvaa asettelu'
+});
diff --git a/src/common/lib/ckeditor/plugins/lineheight/lang/fi.js b/src/common/lib/ckeditor/plugins/lineheight/lang/fi.js
new file mode 100644
index 0000000..5ff63a6
--- /dev/null
+++ b/src/common/lib/ckeditor/plugins/lineheight/lang/fi.js
@@ -0,0 +1,3 @@
+CKEDITOR.plugins.setLang('lineheight','fi', {
+    title: 'Line Height'
+} );
