Index: src/common/lib/ckeditor/lang/bg.js
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
--- src/common/lib/ckeditor/lang/bg.js	(date 1518160009000)
+++ src/common/lib/ckeditor/lang/bg.js	(revision )
@@ -190,7 +190,7 @@
     "toAnchor": "Връзка към котва в текста",
     "toEmail": "E-mail",
     "toUrl": "Уеб адрес",
-    "toolbar": "Връзка",
+    "toolbar": "Вмъкни външна връзка",
     "type": "Тип на връзката",
     "unlink": "Премахни връзката",
     "upload": "Качване"
Index: src/common/lib/ckeditor/lang/en.js
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
--- src/common/lib/ckeditor/lang/en.js	(date 1518160009000)
+++ src/common/lib/ckeditor/lang/en.js	(revision )
@@ -190,7 +190,7 @@
     "toAnchor": "Link to anchor in the text",
     "toEmail": "E-mail",
     "toUrl": "URL",
-    "toolbar": "Link",
+    "toolbar": "Insert external link",
     "type": "Link Type",
     "unlink": "Unlink",
     "upload": "Upload"
Index: src/common/lib/ckeditor/lang/de.js
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
--- src/common/lib/ckeditor/lang/de.js	(date 1518160009000)
+++ src/common/lib/ckeditor/lang/de.js	(revision )
@@ -190,7 +190,7 @@
     "toAnchor": "Anker in dieser Seite",
     "toEmail": "E-Mail",
     "toUrl": "URL",
-    "toolbar": "Link einfügen/editieren",
+    "toolbar": "Einfügen еxternen link",
     "type": "Link-Typ",
     "unlink": "Link entfernen",
     "upload": "Hochladen"
