Index: src/common/lib/ckeditor/plugins/tableselection/plugin.js
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
--- src/common/lib/ckeditor/plugins/tableselection/plugin.js	(date 1517235935000)
+++ src/common/lib/ckeditor/plugins/tableselection/plugin.js	(revision )
@@ -359,9 +359,8 @@
       return;
     }

-    // We're not supporting dragging in our table selection for the time being.
-    evt.cancel();
-    evt.data.preventDefault();
+    //Disable fakeTableSelection on items drag and drop.
+    fakeSelection = { active: false };
   }

   function copyTable( editor, isCut ) {
