diff --git a/src/common/lib/lazyload/jquery.lazyload.js b/src/common/lib/lazyload/jquery.lazyload.js
index ae49e6b..09f3a4c 100644
--- a/src/common/lib/lazyload/jquery.lazyload.js
+++ b/src/common/lib/lazyload/jquery.lazyload.js
@@ -35,6 +35,8 @@
         function update() {
             var counter = 0;
 
+            var elementsLength = elements.length;
+
             elements.each(function() {
                 var $this = $(this);
                 if (settings.skip_invisible && !$this.is(":visible")) {
@@ -49,7 +51,8 @@
                         /* if we found an image we'll load, reset the counter */
                         counter = 0;
                 } else {
-                    if (++counter > settings.failure_limit) {
+                    // iterate each element at least once. Otherwise it will cancel update on the first "out of the view" element
+                    if (++counter > settings.failure_limit + elementsLength) {
                         return false;
                     }
                 }
